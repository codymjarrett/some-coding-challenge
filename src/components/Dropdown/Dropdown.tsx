import React, { ChangeEvent } from 'react'

export interface DropdownProps {
	sortMovies: any
}
const Dropdown: React.SFC<DropdownProps> = ({ sortMovies }) => {
	return (
		<div className="dropdown">
			<div className="dropdown__content">
				<p className="dropdown__content-title">Sort</p>
				<div className="select-wrapper">
					<select
						onChange={(ev: ChangeEvent<HTMLSelectElement>): void =>
							sortMovies(ev.target.value)
						}
					>
						<option value="ASC">Ascending</option>
						<option value="DSC">Descending</option>
					</select>
				</div>
			</div>
		</div>
	)
}

export default Dropdown
