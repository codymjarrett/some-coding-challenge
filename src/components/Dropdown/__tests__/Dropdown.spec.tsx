import React from 'react'
import { render, cleanup } from '@testing-library/react';

import Dropdown from '../Dropdown'

afterEach(cleanup);

it("renders", () => {
  const { asFragment } = render(<Dropdown/>);
  expect(asFragment()).toMatchSnapshot();
});

