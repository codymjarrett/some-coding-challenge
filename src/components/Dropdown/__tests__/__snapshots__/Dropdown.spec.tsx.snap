// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <div
    class="dropdown"
  >
    <div
      class="dropdown__content"
    >
      <p
        class="dropdown__content-title"
      >
        Sort
      </p>
      <div
        class="select-wrapper"
      >
        <select>
          <option
            value="ASC"
          >
            Ascending
          </option>
          <option
            value="DSC"
          >
            Descending
          </option>
        </select>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
